google/cloud/storage/blob.py

```diff
@@ -3092,22 +3093,23 @@ def compose(
         if self.user_project is not None:
             query_params["userProject"] = self.user_project
 
+        _add_generation_match_parameters(
+            query_params,
+            if_generation_match=if_generation_match,
+            if_metageneration_match=if_metageneration_match,
+        )
+
```

Comment:
Given the change proposed above, and that `index` is used only to find the corresponding entry in `if_source_generation_match`, this would be clearer using the `zip` built-in, e.g.:

```python
        for source, source_generation in zip(sources, if_source_generation_match):
```